### DISCUSSION TITLE
Problems with OTP implementation.

### DISCUSSION THREAD

**Discussion by LeCyreaxYT** [pocketbase/pocketbase#5919] [Q&A] [answered]
Hi,
I want to set up OTP for my system so that an admin can send an OTP to the user. The user receives a link to the OTP password page, where they can enter the code to log in.

The problem arises when I execute the following code:
```typescript
const req = await pb.collection('users').requestOTP('test@example.com');
```

The `req` variable contains an `otpId`. However, the OTP entering page does not have access to this otpId because the user does not request the OTP directly.

Is there a way to implement OTP functionality securely, without making the `users` table public or compromising security?
```typescript
const authData = await pb.collection('users').authWithOTP(
    req.otpId,
    "YOUR_OTP",
);
```

Thanks in advance for your help!

**Comment by ganigeorgiev** [comment, marked as answer]
> The user receives a link to the OTP password page, where they can enter the code to log in.

You have access to both the `{OTP}` and `{OTP_ID}` placeholder params that you can use in your email template, aka. you can append `?otpId={OTP_ID}` to your email url and then when the user visits the page read the query params and pass it alongside the auth request.

**Reply by LeCyreaxYT** [reply]
Thanks <3
